src/classes/RD2_DataMigration_TEST.cls

```diff
@@ -1932,6 +1932,55 @@ private class RD2_DataMigration_TEST {
             'Number of Planned Installments should be set to null');
     }
 
+    /**
+     * @description Verifies RD is successfully converted when Amount can be calculated for an
+     * RD with Open Ended Status set to Closed
+     */
+    @IsTest
+    private static void shouldConvertWhenOpenEndedStatusClosedAmountCanBeCalculated() {
+        npe03__Recurring_Donation__c rd = getLegacyRecurringDonationBuilder()
+            .withOpenEndedStatusClosed()
+            .withPlannedInstallments(1)
+            .withDateEstablished(DATE_ESTABLISHED)
+            .build();
+
+        npe03__Recurring_Donation__c convertedRD = new RD2_DataMigrationMapper(rd)
+            .convertToEnhancedRD();
+        System.assertNotEquals(null, convertedRD, 'RD should be successfully converted');
+
+        rd.npe03__Installments__c = 12;
+        rd.npe03__Schedule_Type__c = RD_Constants.SCHEDULE_TYPE_MULTIPLY_BY;
+        convertedRD = new RD2_DataMigrationMapper(rd)
+            .convertToEnhancedRD();
+        System.assertNotEquals(null, convertedRD, 'RD should be successfully converted');
+    }
+
+    /**
+     * @description Verifies Exception is thrown when Amount cannot be calculated for an
+     * RD with Open Ended Status set to Closed
+     */
+    @IsTest
+    private static void shouldThrowExceptionWhenOpenEndedStatusClosedAmountCannotBeCalculated() {
+        npe03__Recurring_Donation__c rd = getLegacyRecurringDonationBuilder()
+            .withOpenEndedStatusClosed()
+            .withPlannedInstallments(12)
+            .withDateEstablished(DATE_ESTABLISHED)
+            .build();
+
+        npe03__Recurring_Donation__c convertedRD;
+        try {
+            convertedRD = new RD2_DataMigrationMapper(rd)
+                .convertToEnhancedRD();
+        }
+        catch (RD2_DataMigrationMapper.MigrationException e) {
+            System.assertEquals(true,
```

Comment:
Let's check exceptions like we usually do with checking an `actualException` after the `try/catch` block.